### PR TITLE
[GenericFilter] Support custom conditions and configurations

### PR DESCRIPTION
**Pull request by KremnevDmitry**
[GenericFilter] Support custom conditions jmix-framework/jmix#1444
[GenericFilter] Support configurations jmix-framework/jmix#1445

### CODE REVIEW

**Review by glebfox — CHANGES_REQUESTED** [review]

File: jmix-flowui/flowui/src/main/java/io/jmix/flowui/component/logicalfilter/LogicalFilterComponent.java [hunk 1]
```diff
@@ -101,9 +104,24 @@
     /**
      * Operation representing corresponding logical filtering condition.
      */
-    enum Operation {
+    enum Operation implements EnumClass<String> {
```
[the remark this answers is not in the document]
Comment (KremnevDmitry) [reply jmix-flowui/flowui/src/main/java/io/jmix/flowui/component/logicalfilter/LogicalFilterComponent.java:107]: I suggest not making any changes.

File: jmix-flowui/flowui/src/main/java/io/jmix/flowui/component/propertyfilter/PropertyFilter.java [hunk 2]
```diff
@@ -368,7 +370,7 @@ public void actionPerform(Component component) {
     /**
      * Operation representing corresponding filtering condition.
      */
-    public enum Operation {
+    public enum Operation implements EnumClass<String> {
```
[the remark this answers is not in the document]
Comment (KremnevDmitry) [reply jmix-flowui/flowui/src/main/java/io/jmix/flowui/component/propertyfilter/PropertyFilter.java:373]: I suggest not making any changes.

**Comment by KremnevDmitry** [comment]
Drafted

New PR: https://github.com/jmix-framework/jmix/pull/1653

**Comment by KremnevDmitry** [comment]
Closed

New PR: https://github.com/jmix-framework/jmix/pull/1653

**Pull request closed without merging**